great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -1264,6 +1242,38 @@ def delete_profiler(
                 message=f'Non-existent Profiler configuration named "{id}".\n\nDetails: {exc_ik}'
             )
 
+    @staticmethod
+    def update_profiler(
+        profiler: RuleBasedProfiler,
+        profiler_store: ProfilerStore,
+        data_context: AbstractDataContext,
+    ) -> None:
+        name = profiler.name
+        id = profiler.ge_cloud_id
+
+        key = RuleBasedProfiler._construct_profiler_key(name=name, id=id)
+        if not profiler_store.has_key(key):
+            msg = "Non-existent Profiler configuration "
+            if name:
+                msg += f"named {name}"
+            elif id:
+                msg += f"with id {id}"
+            raise gx_exceptions.ProfilerNotFoundError(message=msg)
+
+        # Prioritize deletion by id if present (providing both will result in a downstream assertion error)
+        if id:
+            RuleBasedProfiler.delete_profiler(
+                ge_cloud_id=id, profiler_store=profiler_store
+            )
+        else:
+            RuleBasedProfiler.delete_profiler(name=name, profiler_store=profiler_store)
+
+        _ = RuleBasedProfiler.add_profiler(
```

Comment:
`add_profiler` returns a `RuleBasedProfiler`, but `update_profiler` does not. Why is that?

Reply:
none of our `update` methods actually return an object - since the update call takes in an instance of an RBP to use to update the value persisted in the store, we already have that object for downstream usage.

```python

profiler = RBP(...)
profiler.rules = {...}

context.update_profiler(profiler)
# Since `profiler` has my updates, do I need a return value for my `update_profiler` call?
```